### PR TITLE
Add BitmapIterationBenchmark and update to JMH 1.17.2

### PR DESCRIPTION
**Pull request by leventov**
- Update to the bleeding edge JMH version (1.17.2)
 - Add `BitmapIterationBenchmark`, a collection of benchmarks of bitmap iteration and iteration + something (cumulative cost), the latter is useful for comparing total "usage cost" of different `BitmapSerdeFactory` impls (bitmap algorithms).

Results (on a noisy consumer-level laptop, so double-check them on your production hardware, but useful anyway):
```
Benchmark                                     (bitmapAlgo)  (n)  (prob)   (size)  Mode  Cnt          Score          Error  Units
BitmapIterationBenchmark.constructAndIter           bitset  N/A     0.5  1000000  avgt    5    3338291.030 ±   180233.306  ns/op
BitmapIterationBenchmark.constructAndIter          concise  N/A     0.5  1000000  avgt    5   16496743.655 ±   545798.361  ns/op
BitmapIterationBenchmark.constructAndIter          roaring  N/A     0.5  1000000  avgt    5    8448985.138 ±   336566.515  ns/op
BitmapIterationBenchmark.intersectionAndIter        bitset    2     0.5  1000000  avgt    5    2957560.842 ±    39086.454  ns/op
BitmapIterationBenchmark.intersectionAndIter        bitset   10     0.5  1000000  avgt    5    3999658.647 ±   113291.313  ns/op
BitmapIterationBenchmark.intersectionAndIter        bitset  100     0.5  1000000  avgt    5    6710492.009 ±   222596.163  ns/op
BitmapIterationBenchmark.intersectionAndIter       concise    2     0.5  1000000  avgt    5    9036510.079 ±  1183236.075  ns/op
BitmapIterationBenchmark.intersectionAndIter       concise   10     0.5  1000000  avgt    5   12763239.333 ±  1052581.264  ns/op
BitmapIterationBenchmark.intersectionAndIter       concise  100     0.5  1000000  avgt    5  134267030.500 ±  9484080.283  ns/op
BitmapIterationBenchmark.intersectionAndIter       roaring    2     0.5  1000000  avgt    5    1717210.820 ±   131885.259  ns/op
BitmapIterationBenchmark.intersectionAndIter       roaring   10     0.5  1000000  avgt    5    1866879.017 ±   135501.817  ns/op
BitmapIterationBenchmark.intersectionAndIter       roaring  100     0.5  1000000  avgt    5    3583572.984 ±    88640.480  ns/op
BitmapIterationBenchmark.iter                       bitset  N/A     0.5  1000000  avgt    5    2056744.108 ±    15253.863  ns/op
BitmapIterationBenchmark.iter                      concise  N/A     0.5  1000000  avgt    5    7676221.277 ±   223318.985  ns/op
BitmapIterationBenchmark.iter                      roaring  N/A     0.5  1000000  avgt    5    1679534.873 ±   100423.234  ns/op
BitmapIterationBenchmark.unionAndIter               bitset    2     0.5  1000000  avgt    5    1556912.050 ±    76009.537  ns/op
BitmapIterationBenchmark.unionAndIter               bitset   10     0.5  1000000  avgt    5     748860.193 ±    62327.144  ns/op
BitmapIterationBenchmark.unionAndIter               bitset  100     0.5  1000000  avgt    5    4683472.828 ±   246870.714  ns/op
BitmapIterationBenchmark.unionAndIter              concise    2     0.5  1000000  avgt    5    8372220.568 ±  1835927.350  ns/op
BitmapIterationBenchmark.unionAndIter              concise   10     0.5  1000000  avgt    5   11329175.926 ±  1000462.128  ns/op
BitmapIterationBenchmark.unionAndIter              concise  100     0.5  1000000  avgt    5  164556330.171 ± 10902682.709  ns/op
BitmapIterationBenchmark.unionAndIter              roaring    2     0.5  1000000  avgt    5    2375032.871 ±   172506.529  ns/op
BitmapIterationBenchmark.unionAndIter              roaring   10     0.5  1000000  avgt    5    4044270.278 ±  1375682.967  ns/op
BitmapIterationBenchmark.unionAndIter              roaring  100     0.5  1000000  avgt    5    6306893.787 ±  1778120.660  ns/op
```

TLDR: roaring is orders of magnitude faster than concise, let's make it the default algorithm.

### CODE REVIEW

**Comment by gianm** [comment]
@leventov fwiw, the two main reasons we haven't made Roaring default in the past are:

- reports that compressed bitmap size is larger than Concise on various datasets
- bugs in the early implementation (although bug reports are almost zero these days, and many people are using Roaring in production with Druid, so this is less of an issue now)

**Comment by gianm** [comment]
But yes I think all Druid users that have complex filtering needs should consider Roaring.

**Comment by fjy** [comment]
👍
